### PR TITLE
Fix #271: Add job history details

### LINKED ISSUE #271: Create view with jobs histories
There have been several jobs failing silently. Those jobs will soon generate alerts (#201) but it would  still be convenient to have a master view in the dashboard that shows the history, and their status, of all scheduled jobs.  Furthermore, every user should be able to see the history for their own jobs.

### PR DESCRIPTION
**Pull request by robhudson**
screenshot 2017-05-17 17 06 51

### CODE REVIEW

**Comment by robhudson** [comment]
What's in the table is basically what's in the spark job runs table.

Some questions:
* Do we need pagination when this table gets large?
* Are there other columns you can think of?

**Comment by mreid-moz** [comment]
> Are there other columns you can think of?

If feasible, a link to the logs would be great. Also the cluster size, in case the user updates it.

**Comment by robhudson** [comment]
> If feasible, a link to the logs would be great. Also the cluster size, in case the user updates it.

Right now we don't store the logs with the run data we have. The list of logs in the logs pane is simply a listing of the S3 bucket where the logs are stored. But, this could be changed so we can associated exactly which logs apply to which run. Since the logs have a date stamp in them it's probably pretty easy to map them together but I think it'd be nice to relate them on the backend as well.

As far as cluster size we're currently only tracking that at the job level, not for each run so we don't have that data. But again, we can change it on the backend if that's what we want to do.

**Comment by robhudson** [comment]
I filed issues for the logs and cluster size.

**Comment by mreid-moz** [comment]
Thanks! Both are just nice-to-haves, so a separate followup issue is fine w/ me.

File: atmo/templates/atmo/jobs/detail.html [hunk 1]
```diff
@@ -163,6 +164,36 @@
...
+            </thead>
+            <tbody>
+              {% for run in spark_job.runs.all %}
+              <tr>
+                <td>{{ run.jobflow_id }}</a></td>
+                <td>{{ run.emr_release_version }}</td>
+                <td>{{ run.status }}</td>
+                <td>{{ run.scheduled_date|default:"n/a" }}</td>
+                <td>{{ run.run_date|default:"n/a" }}</td>
+                <td>{{ run.terminated_date|default:"n/a" }}</td>
```
Comment (jezdez) [inline atmo/templates/atmo/jobs/detail.html]: This will require an update when #481 is merged.

File: atmo/templates/atmo/jobs/detail.html [hunk 2]
```diff
@@ -163,6 +164,36 @@
...
+                <th>Last status</th>
+                <th>Scheduled</th>
+                <th>Ran</th>
+                <th>Terminated</th>
+              </tr>
+            </thead>
+            <tbody>
+              {% for run in spark_job.runs.all %}
+              <tr>
+                <td>{{ run.jobflow_id }}</a></td>
```
Comment (jezdez) [inline atmo/templates/atmo/jobs/detail.html]: It'd be nice to use the Jobflow ID to link to the AWS EMR console if the current user has permissions (is staff? member of "Spark job maintainers" group which exists in prod?).

The URL for a job is `https://<AWS region>.console.aws.amazon.com/elasticmapreduce/home?region=<AWS region>#cluster-details:<Jobflow ID>`, e.g. `https://us-west-2.console.aws.amazon.com/elasticmapreduce/home?region=us-west-2#cluster-details:j-IQROSTWRHXIH`.

Comment (robhudson) [inline atmo/templates/atmo/jobs/detail.html]: Is the AWS region always the same? Do we need to have any permissions for the link? The detail page is using the `@view_permission_required(SparkJob)` and the link requires AWS credentials so if they have those they are good.

**Comment by robhudson** [comment]
Updated with the cluster size and link to AWS EMR console.

File: atmo/templates/atmo/jobs/detail.html [hunk 3]
```diff
@@ -163,6 +164,38 @@
...
+                <th>Last status</th>
+                <th>Scheduled</th>
+                <th>Ran</th>
+                <th>Terminated</th>
+              </tr>
+            </thead>
+            <tbody>
+              {% for run in spark_job.runs.all %}
+              <tr>
+                <td><a href="https://us-west-2.console.aws.amazon.com/elasticmapreduce/home?region=us-west-2#cluster-details:{{ run.jobflow_id }}">{{ run.jobflow_id }}</a></td>
```
Comment (jezdez) [inline atmo/templates/atmo/jobs/detail.html]: Let's not hard code the AWS region here, use this instead:

```
<td><a href="https://{{ settings.AWS_CONFIG.AWS_REGION }}.console.aws.amazon.com/elasticmapreduce/home?region={{ settings.AWS_CONFIG.AWS_REGION }}#cluster-details:{{ run.jobflow_id }}">{{ run.jobflow_id }}</a></td>
```

File: atmo/templates/atmo/jobs/detail.html (hunk 2 above)
Comment (jezdez) [inline atmo/templates/atmo/jobs/detail.html]: @robhudson The region is configured in the `settings.AWS_CONFIG['AWS_REGION']` setting, so let's use that for the value. I made a comment in the review for this.

Comment (jezdez) [inline atmo/templates/atmo/jobs/detail.html]: Oh, and you're right, the permissions suffice.

**Pull request merged**